sklearn/linear_model/_logistic.py

```diff
@@ -506,6 +506,9 @@ def _logistic_regression_path(
                 w0 = np.concatenate([coef_.ravel(), intercept_])
             else:
                 w0 = coef_.ravel()
+            # n_iter_i is an array for each class. However, `target` is always encoded
+            # in {-1, 1}, so we only take the first element of n_iter_i.
+            n_iter_i = n_iter_i[0]
```

Comment:
So here this is really specific to `liblinear`. We should preserve the array because it will be useful in multiclass case in other part of the code.